custom-recipes/send-mails-from-contacts-dataset/recipe.py

```diff
@@ -78,45 +105,76 @@ def send_email_for_contact(mail_client, contacts_row, message_template):
 
 use_body_value = config.get('use_body_value', False)
 
-# For legacy configs, assume it is text if not defined
-use_html_body_value = config.get('body_format', 'text') == 'html'
-
 body_column = config.get('body_column', None)
 body_value = config.get('body_value', None)
-html_body_value = config.get('html_body_value', None)
 
-body_encoding = config.get('body_encoding', 'us-ascii')
+# For legacy configs, assume it is text if not defined
+body_format = config.get('body_format', 'text')
+# For sending a body from a column value we also assume it is plain text - that is the legacy behavour
+use_html_body_value = body_value and (body_format == 'html')
 
-smtp_config = read_smtp_config(config)
+html_body_value = config.get('html_body_value', None)
+mail_channel = config.get('mail_channel', None)
+channel_has_sender = does_channel_have_sender(mail_channel)
 
 attachment_type = config.get('attachment_type', "csv")
 
-# Some validation, check we have things we really need
-if not body_column and not (use_body_value and body_value) and not (use_html_body_value and html_body_value):
+# Validation part 1 - Check some kind of value/column exists for body, subject and sender
+
+has_body_column = body_column and not use_body_value
+has_plain_body_value = use_body_value and body_value and not use_html_body_value
+has_html_body_value = use_body_value and use_html_body_value and html_body_value
+if not (has_body_column or has_plain_body_value or has_html_body_value):
     raise AttributeError("No body column nor body value specified")
 
+has_subject_column = subject_column and not use_subject_value
+has_subject_value = use_subject_value and subject_value
+if not has_subject_column and not has_subject_value:
+    raise AttributeError("No value provided for the subject")
+
+
+has_sender_column = sender_column and not use_sender_value and not channel_has_sender
+has_sender_value = use_sender_value and sender_value and not channel_has_sender
+if not (channel_has_sender or has_sender_column or has_sender_value):
+    raise AttributeError("No value provided for the sender")
```

Comment:
Maybe there should be a similar check for the recipient column

Reply:
Done [here](https://github.com/dataiku/dss-plugin-sendmail/pull/9/commits/87a5861fe84296151b32d79bb4dc8e00ead45de2)